engines/wintermute/ad/ad_scene_geometry.cpp

```diff
@@ -496,74 +528,40 @@ bool AdSceneGeometry::render(bool render) {
 
 //////////////////////////////////////////////////////////////////////////
 bool AdSceneGeometry::renderShadowGeometry() {
-	storeDrawingParams();
-
-	warning("AdSceneGeometry::renderShadowGeometry not yet implemented");
-
-	// implement this later
-
-	//	CBRenderD3D* m_Renderer = (CBRenderD3D*)_gameRef->m_Renderer;
-
-	//	// render the geometry
-	//	Math::Matrix4 matIdentity;
-	//	matIdentity.setToIdentity();
-
-	//	if(m_ActiveCamera>=0 && m_ActiveCamera<_cameras.size())
-	//		m_Renderer->Setup3D(_cameras[m_ActiveCamera]);
-
-	//	m_Renderer->m_Device->SetTransform(D3DTS_WORLD, &matIdentity);
-
-	//	// disable color write
-	//	m_Renderer->SetSpriteBlendMode(BLEND_UNKNOWN);
-	//	m_Renderer->m_Device->SetRenderState(D3DRS_SRCBLEND,  D3DBLEND_ZERO);
-	//	m_Renderer->m_Device->SetRenderState(D3DRS_DESTBLEND, D3DBLEND_ONE);
-
-	//	m_Renderer->m_Device->SetTransform(D3DTS_WORLD, &matIdentity);
-	//	C3DUtils::SetFixedVertexShader(m_Renderer->m_Device, D3DFVF_MODELVERTEXCOLOR);
-
-	//	// no texture
-	//	m_Renderer->m_LastTexture = NULL;
-	//	m_Renderer->m_Device->SetTexture(0, NULL);
-
-	//	m_Renderer->m_Device->SetRenderState(D3DRS_CULLMODE, D3DCULL_CCW);
-
-	//	// render blocks
-	//	for(int i=0; i<_blocks.size(); i++)
-	//	{
-	//		if(!_blocks[i]->m_Active) continue;
-	//		if(!_blocks[i]->m_Mesh->m_VB) continue;
-	//		if(!_blocks[i]->m_ReceiveShadows) continue;
-	//		C3DUtils::SetStreamSource(m_Renderer->m_Device, 0, _blocks[i]->m_Mesh->m_VB, sizeof(MODELVERTEXCOLOR));
-	//		m_Renderer->m_Device->DrawPrimitive(D3DPT_TRIANGLELIST, 0, _blocks[i]->m_Mesh->m_NumFaces);
+	_gameRef->_renderer3D->resetModelViewTransform();
+	_gameRef->_renderer3D->setup3D(getActiveCamera(), true);
 
-	//		m_Renderer->m_NumPolygons += _blocks[i]->m_Mesh->m_NumFaces;
-	//	}
+	// disable color write
+	glBlendFunc(GL_ZERO, GL_ONE);
 
-	//	// render walkplanes
-	//	for(int i=0; i<_planes.size(); i++)
-	//	{
-	//		if(!_planes[i]->m_Active) continue;
-	//		if(!_planes[i]->m_Mesh->m_VB) continue;
-	//		if(!_planes[i]->m_ReceiveShadows) continue;
-	//		C3DUtils::SetStreamSource(m_Renderer->m_Device, 0, _planes[i]->m_Mesh->m_VB, sizeof(MODELVERTEXCOLOR));
-	//		m_Renderer->m_Device->DrawPrimitive(D3DPT_TRIANGLELIST, 0, _planes[i]->m_Mesh->m_NumFaces);
+	glFrontFace(GL_CCW);
+	glBindTexture(GL_TEXTURE_2D, 0);
 
-	//		m_Renderer->m_NumPolygons += _planes[i]->m_Mesh->m_NumFaces;
-	//	}
+	// render walk planes
+	for (uint i = 0; i < _planes.size(); i++) {
+		if (_planes[i]->_active && _planes[i]->_receiveShadows) {
+			_planes[i]->_mesh->render();
+			//m_Renderer->m_NumPolygons += _planes[i]->m_Mesh->m_NumFaces;
```

Comment:
Why is this commented out?

Reply:
The member variable is not part of the renderer yet and I don't even know, what it's use is. So this should probably stay as a reminder.